tests/unit/policy_test.py

```diff
@@ -892,58 +892,116 @@ def test_surface_policy_moves_back_to_object(self):
             for loader_step, observation in enumerate(exp.dataloader):
                 exp.model.step(observation)
 
-                # |Step| Action           | Motor-only? | Obs processed? | Source
-                # |----|------------------|-------------|----------------|-------------
-                # | 13 | OrientHorizontal | True        | False          | touch_object
-                # | 14 | OrientHorizontal | True        | False          | touch_object
-                # | 15 | OrientHorizontal | True        | False          | touch_object
-                # | 16 | OrientHorizontal | True        | False          | touch_object
-                # | 17 | OrientHorizontal | True        | False          | touch_object
-                # | 18 | OrientHorizontal | True        | False          | touch_object
-                # | 19 | OrientHorizontal | True        | False          | touch_object
-                # | 20 | OrientHorizontal | True        | False          | touch_object
-                # | 21 | OrientHorizontal | True        | False          | touch_object
-                # | 22 | OrientHorizontal | True        | False          | touch_object
-                # | 23 | OrientHorizontal | True        | False          | touch_object
-                # | 24 | OrientHorizontal | True        | False          | touch_object
-                # | 25 | OrientVertical   | True        | False          | touch_object
-                # | 26 | MoveTangentially | True        | False          | dynamic_call
-                # | 27 | OrientVertical   | True        | False          | touch_object
-                # | 28 | MoveForward      | True        | False          | touch_object
-                # | 29 | OrientHorizontal | True        | False          | dynamic_call
-                # | 30 | OrientVertical   | True        | True           | dynamic_call
-                #
-                # Note that 26 (MoveTangentially) is a bug, in that SurfacePolicy
-                # dynamic_call assumes that it is in the middle of the
-                # MoveForward->OrientHorizontal->OrientVertical->MoveTangentially
-                # cycle, but in fact it is not.
-                # As a result, the sensor falls off the object again.
-
-                if 13 <= loader_step <= 25:  # Motor-only touch_object steps
+                #  Step | Action           | Motor-only? | Obs processed? | Source
```

Comment:
Are motor_only and obs_processed ever both True or both False?  Do we need both columns here?

Reply:
> Are motor_only and obs_processed ever both True or both False?

Yes, when a sensor thinks it should be sending data to the learning module `(OrientVertical,motor_only_step=False`), but observes no data, the result will be `motor_only_step=False` and `obs_processed=False`.

Reply:
I will rework this test and this event flow in the following pull request when I implement the `TouchObject` positioning procedure.